### ISSUE TITLE
Nvidia Drivers disables Wayland on GNOME-43

### ISSUE THREAD

**Issue by Neon-44** [NixOS/nixpkgs#203355]
### Describe the bug
when i import my Nvidia Configuration (hardware.opengl.enable + services.xserver.videoDrivers = "nvidia"), i can't enable Wayland anymore in the Login Screen. the small gear in the bottom right corner just disappears.

### Steps, To Reproduce
1. copy gnome config from NixOS website
2. copy Nvidia Config from NixOS Website, grey out the optional part
3. put both configs into seperate files
4. import files from main config
5. build config from flake with nixpkgs = unstable channel
6. gear disappears in login screen
optional:
7. disable nvidia config import
8. be able to enable wayland in gear in login screen

### Expected behavior
i should still be able to use Wayland when using the proprietary Nvidia Drivers

### Additional context
i'm pretty sure it still worked a few days ago. i was able to once log in using wayland, but then not anymore afterwards.
i'm pretty sure a update broke it, because i didn't touch those configs anymore

### Notify maintainers

am not too deep into git, so i couldn't find out where to find anyone, but it seems to be services.xserver.videoDrivers

### Metadata

sorry, i had to redact my username as it is also my full Name

<fullname>@nixos-Laptop-<firstname> /n/c/drivers (master)> nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.79, NixOS, 22.11 (Raccoon), 22.11.20221121.20fc948`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.0`
 - channels(<fullname>): `""`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

**Comment by buckley310** [comment]
Same behavior on my Nvidia Optimus machine.
Related https://bugzilla.redhat.com/show_bug.cgi?id=2128910

**Comment by JanKaifer** [comment]
Using `noveau` drivers fixed it for me.
https://github.com/JanKaifer/nixos/blob/15f38b686f309cfaa3a01e8b545e29e8b6172af0/machines/pearbox/configuration.nix#L24

**Comment by Neon-44** [comment]
yes, nouevau Drivers work, but they don't go well with my ultra-wide monitor.

they just cut off the right third of my Monitor as if it was a 16:9 display

**Comment by buckley310** [comment]
Have you tried the `NVreg_PreserveVideoMemoryAllocations` setting mentioned in the redhat bugzilla? My system is also affected by this so I can try it eventually, just haven't yet.

**Comment by jupblb** [comment]
I also have this problem.

**Comment by Neon-44** [comment]
has anyone of you guys been able to fix this by now?

**Comment by Neon-44** [comment]
i managed to somewhat (hot)fix it:

```
  boot.kernelParams = ["nvidia_drm.modeset=1"];
```
brings the option back, but it's a lot jankier than i remember it and defenitely a lot jankier than on my AMD Laptop
nixpkg VSCode i.e. is now just a white window that can't even be resized (weird graphical glitches), and Flatpak VSCode isn't perfect either
so use at your own risk
but hey, the button is back at least. maybe (hopefully) this will help the Team locate and fix the Bug (and thereby the Jankieness) :)

**Comment by Xbz-24** [comment]
did someone find any solution to this? my wayland gets disabled when I try to use nvidia drivers, also when setting
hardware.nvidia.modesetting.enable = true;
I dont see my cursor and hyprland starts acting glitchy, my problem is severe since I am developing a game in SFML and the code compiles and runs in my mbp but when I try to compile it in NixOS I get a segmentation fault when creating the screen

sf::RenderWindow window(sf::VideoMode::getDesktopMode(), "Pong", sf::Style::Default, settings);

the code runs and compiles properly in macOS but when I try to reproduce and play my game in my NixOS enviroment with wayland and xserver somehow I get a segmentation fault in this line, but according to the guys from the SFML discord server its properly defined (they checked all my code)

**Comment by buckley310** [comment]
> hyprland starts acting glitchy

> my NixOS enviroment with wayland and xserver

> I tried the noveau drivers

This issue is a GDM+GNOME+Nvidia specific problem, so it sounds like you're running into a different issue.
You could start a new github issue to debug that problem, though.
